direnv loses file watching when devenv build fails

Currently whenever devenv is not able to build or not able to evaluate, a Nix error message is shown without context for user what this means. When editing `devenv.nix` with (for instance) vim and fixing the issue, direnv won't automatically load devenv anymore: the file watching has been disabled because `.envrc` has exited with code 1.

Example:

```console
$ cd myproject
...
       > make[1]: Leaving directory '/build/source/pjsip/build'
       > make: *** [Makefile:14: all] Error 1
       For full logs, run 'nix log /nix/store/pmfkniw9255dwrfhwhpz7ldwgnjkklj4-pjsip-2.13.drv'.
error:
       error: 1 dependencies of derivation '/nix/store/i67zr8avdnj4hsb320lya99jwybg9p1v-devenv-shell-env.drv' failed to build
$ vim devenv.nix
$ 
# Still no devenv
```

It would be nice if direnv would still be intact and was still watching the devenv files for a rerun of the build.
In addition, it would be nice to explain to the user that the build of devenv has failed and that the devenv environment is not available.

--- a/templates/simple/.envrc
+++ b/templates/simple/.envrc
@@ -5,4 +5,7 @@ fi
 nix_direnv_watch_file devenv.nix
 nix_direnv_watch_file devenv.lock
 nix_direnv_watch_file devenv.yaml
-use flake . --impure
+if ! use flake . --impure
+then
+  echo "devenv could not be build. The devenv environment was not loaded. Make the necessary changes to devenv.nix and hit enter to try again." >&2
+fi

